core/tests/ts-integration/src/modifiers/balance-checker.ts

```diff
@@ -194,44 +197,75 @@ class ShouldChangeBalance extends MatcherModifier {
  *
  * @param receipt Receipt of the transaction to extract fee from.
  * @param from Optional substitute to `receipt.from`.
+ * @param wallet Optional wallet if the api mode is modern
  * @returns Extracted fee
  */
-export function extractFee(receipt: zksync.types.TransactionReceipt, from?: string): Fee {
+export async function extractFee(
+    receipt: zksync.types.TransactionReceipt,
+    from?: string,
+    wallet?: zksync.Wallet
+): Promise<Fee> {
     from = from ?? receipt.from;
 
     const systemAccountAddress = '0x0000000000000000000000000000000000000000000000000000000000008001';
     // We need to pad address to represent 256-bit value.
     const fromAccountAddress = ethers.utils.hexZeroPad(ethers.utils.arrayify(from), 32);
-    // Fee log is one that sends money to the system contract account.
-    const feeLog = receipt.logs.find((log) => {
-        return log.topics.length == 3 && log.topics[1] == fromAccountAddress && log.topics[2] == systemAccountAddress;
-    });
-    if (!feeLog) {
-        throw {
-            message: `No fee log was found in the following transaction receipt`,
-            receipt
-        };
-    }
 
-    const feeAmount = ethers.BigNumber.from(feeLog.data);
+    const apiMode = process.env.EN_API_MODE ?? process.env.API_WEB3_JSON_RPC_API_MODE;
```

Comment:
This kind of implicit checks often becomes a huge PITA to fix, so I'd be extremely cautious about it and generally avoid at all costs.